semantic-conventions/src/opentelemetry/semconv/model/semantic_convention.py

```diff
@@ -53,6 +53,32 @@ def parse(span_kind_value):
         return kind_map.get(span_kind_value)
 
 
+class InstrumentKind(Enum):
+    COUNTER = 1
+    UP_DOWN_COUNTER = 2
+    HISTOGRAM = 3
+    GAUGE = 4
+
+    @staticmethod
+    def parse(instrument_kind_value):
+        return InstrumentKind.kind_map().get(instrument_kind_value)
+
+    @staticmethod
+    def kind_map():
+        return {
+            "Counter": InstrumentKind.COUNTER,
+            "UpDownCounter": InstrumentKind.UP_DOWN_COUNTER,
+            "Histogram": InstrumentKind.HISTOGRAM,
+            "Gauge": InstrumentKind.GAUGE,
+        }
+
+    def __str__(self):
+        # reverse lookup kind_map()
+        return next(filter(lambda i: i[1] is self, InstrumentKind.kind_map().items()))[
```

Comment:
I would not override `str` for enums. What is it used for?

Reply:
I've gone ahead and removed this entire class. I had to do this to correlate the ENUM_NAME with the YamlName. I decided Enum is not a good fit and decided InstrumentKind doesn't need it's own type. 

It can just be a string and we check if it's allowed instrument strings.